File: calculator.py
class Calculator:
    def __init__(self):
        self.history = []

    def add(self, x, y):
        result = x + y
        self.history.append(f"{x} + {y} = {result}")
        return result

    def subtract(self, x, y):
        result = x - y
        self.history.append(f"{x} - {y} = {result}")
        return result

    def multiply(self, x, y):
        result = x * y
        self.history.append(f"{x} * {y} = {result}")
        return result

    def divide(self, x, y):
        if y == 0:
            return "Error: Division by zero"
        result = x / y
        self.history.append(f"{x} / {y} = {result}")
        return result

    def square_root(self, x):
        if x < 0:
            return "Error: Cannot take square root of a negative number"
        result = x ** 0.5
        self.history.append(f"sqrt({x}) = {result}")
        return result

    def exponent(self, base, exp):
        result = base ** exp
        self.history.append(f"{base} ^ {exp} = {result}")
        return result

    def get_history(self):
        return self.history

# Example usage
if __name__ == '__main__':
    calc = Calculator()
    print(calc.add(5, 3))  # Output: 8
    print(calc.subtract(10, 4))  # Output: 6
    print(calc.multiply(2, 6))  # Output: 12
    print(calc.divide(8, 2))  # Output: 4.0
    print(calc.square_root(16))  # Output: 4.0
    print(calc.exponent(2, 3))  # Output: 8
    print(calc.get_history())

File: calculator.js
class Calculator {
  constructor() {
    this.history = [];
  }

  add(a, b) {
    const result = a + b;
    this.history.push(`${a} + ${b} = ${result}`);
    return result;
  }

  subtract(a, b) {
    const result = a - b;
    this.history.push(`${a} - ${b} = ${result}`);
    return result;
  }

  multiply(a, b) {
    const result = a * b;
    this.history.push(`${a} * ${b} = ${result}`);
    return result;
  }

  divide(a, b) {
    if (b === 0) {
      return "Error: Division by zero";
    }
    const result = a / b;
    this.history.push(`${a} / ${b} = ${result}`);
    return result;
  }

  squareRoot(x) {
    if (x < 0) {
      return "Error: Cannot take square root of a negative number";
    }
    const result = Math.sqrt(x);
    this.history.push(`sqrt(${x}) = ${result}`);
    return result;
  }

  power(base, exponent) {
    const result = Math.pow(base, exponent);
    this.history.push(`${base} ^ ${exponent} = ${result}`);
    return result;
  }

  getHistory() {
    return this.history;
  }

  clearHistory() {
    this.history = [];
  }

  printHistory() {
    console.log("--- Calculation History ---");
    this.history.forEach((operation, index) => {
      console.log(`${index + 1}. ${operation}`);
    });
    console.log("---------------------------");
  }
}

// Example usage
const calc = new Calculator();
console.log(calc.add(5, 3));           // Output: 8
console.log(calc.subtract(10, 4));     // Output: 6
console.log(calc.multiply(2, 6));      // Output: 12
console.log(calc.divide(8, 2));        // Output: 4
console.log(calc.squareRoot(16));      // Output: 4
console.log(calc.power(2, 3));         // Output: 8
calc.printHistory();

// Export for use as a module
if (typeof module !== 'undefined' && module.exports) {
  module.exports = Calculator;
}